### PR TITLE
Fix discovery rule priority tests

### PR DESCRIPTION
**Pull request by Gauravtalreja1**

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_discoveryrule.py

File: tests/foreman/cli/test_discoveryrule.py [hunk 1]
```diff
@@ -222,8 +222,8 @@ def test_positive_create_with_priority(self, discoveryrule_factory):
 
         :CaseImportance: Critical
         """
-        available = set(range(1, 1000)) - {Box(r).priority for r in DiscoveryRule.list()}
-        rule_priority = random.sample(available, 1)
+        available = set(range(1, 1000)) - {int(Box(r).priority) for r in DiscoveryRule.list()}
+        rule_priority = random.sample(sorted(available), 1)
```
Comment (Griffin-Sullivan) [inline tests/foreman/cli/test_discoveryrule.py:226]: Why would we sort the set?

**Review by Griffin-Sullivan — APPROVED** [review]
ACK pending PRT. I do remember these tests occasionally failing, but I could never reproduce locally. Casting to an int should help.

File: tests/foreman/cli/test_discoveryrule.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_discoveryrule.py:226]: I just followed the error`E   TypeError: Population must be a sequence.  For dicts or sets, use sorted(d).` :)

Comment (Griffin-Sullivan) [reply tests/foreman/cli/test_discoveryrule.py:226]: Hmmm ok I thought this was a different error. I used to sometimes get a BoxKeyError that would say the rule coming out of `DiscoveryRule.list()` had no priority attribute.

Comment (adarshdubey-star) [reply tests/foreman/cli/test_discoveryrule.py:226]: Good catch with the error @Gauravtalreja1

**Review by adarshdubey-star — APPROVED** [review]
Ack

**Review by shubhamsg199 — APPROVED** [review]
ACK

**Pull request merged**